testsuite/testcases/src/validator_join_leave_test.rs

```diff
@@ -0,0 +1,257 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::generate_traffic;
+use aptos::account::create::DEFAULT_FUNDED_COINS;
+use aptos_genesis::config::HostAndPort;
+use aptos_logger::info;
+use aptos_sdk::crypto::ed25519::Ed25519PrivateKey;
+use aptos_sdk::crypto::{bls12381, x25519};
+use forge::{
+    reconfig, NetworkContext, NetworkTest, NodeExt, Result, SwarmExt, Test, FORGE_KEY_SEED,
+};
+
+use aptos_keygen::KeyGen;
+
+use aptos::test::CliTestFramework;
+use aptos_sdk::types::network_address::DnsName;
+use std::time::Duration;
+use tokio::runtime::Runtime;
+
+const MAX_NODE_LAG_SECS: u64 = 10;
+
+pub struct ValidatorJoinLeaveTest;
+
+impl Test for ValidatorJoinLeaveTest {
+    fn name(&self) -> &'static str {
+        "validator join and leave sets"
+    }
+}
+
+impl NetworkTest for ValidatorJoinLeaveTest {
+    fn run<'t>(&self, ctx: &mut NetworkContext<'t>) -> Result<()> {
+        // Verify we have at least 7 validators (i.e., 3f+1, where f is 2)
```

Comment:
Instead of only 7, let's increase this to something like 20, so that we can have more number of validators leaving and joining at the same time.

Reply:
Done